apollo-ios/Sources/Apollo/ApolloClient.swift

```diff
@@ -1,51 +1,72 @@
-import Foundation
 import Dispatch
+import Foundation
+
 #if !COCOAPODS
-import ApolloAPI
+  import ApolloAPI
 #endif
 
 /// A cache policy that specifies whether results should be fetched from the server or loaded from the local cache.
-#warning("TODO: rethink this API. Onlye valid for queries currently")
+#warning("TODO: rethink this API. Only valid for queries currently")
 public enum CachePolicy: Sendable, Hashable {
   /// Return data from the cache if available, else fetch results from the server.
-  case returnCacheDataElseFetch
-  ///  Always fetch results from the server.
-  case fetchIgnoringCacheData
-  ///  Always fetch results from the server, and don't store these in the cache.
-  case fetchIgnoringCacheCompletely
-  /// Return data from the cache if available, else return an error.
-  case returnCacheDataDontFetch
+  case cacheFirst
```

Comment:
I wanted to comment that `cacheFirst` sounds like a different default behaviour than `returnCacheDataElseFetch` but on second thought I don't think the intent of the cache policy has actually changed, the documentation for the case is still the same. That leads me to think that we need to consider another name.
* `returnCacheDataElseFetch` is mutually exclusive, i.e.: one or the other will happen, not both.
* `cacheFirst` sounds like there is a secondary action, i.e.: cache first, then network.

Reply:
If the case name changes from this then we should take a second look at the rest too.

Reply:
That's fair. For the work I'm doing right now, I have predefined `FetchBehavior` values called `CacheThenNetwork` and `CacheElseNetwork`. I'm open to changing this to `.cacheElseNetwork.

The reason why I went with `cacheFirst` and `networkFirst` was to align with the [cache policy definitions in ApolloKotlin](https://apollographql.github.io/apollo-kotlin/kdoc/older/4.2.0/apollo-http-cache/com.apollographql.apollo.cache.http/-http-fetch-policy/index.html).

@martinbonnin Do you think that `cacheFirst` has been confusing to users in any way?

Reply:
I decided to go with something more specific and clear for all of the cache policy values. This is now implemented in #684.

Reply:
@AnthonyMDev we haven't had any issue with [`CacheFirst`](https://apollographql.github.io/apollo-kotlin/kdoc/older/4.2.0/apollo-normalized-cache/com.apollographql.apollo.cache.normalized/-fetch-policy/-cache-first/index.html). This is also what [Apollo Client is using](https://www.apollographql.com/docs/react/data/queries#supported-fetch-policies) and I think quite established today. If you can, I'd love to get those aligned between different platforms. 

Ping @BoD

Reply:
Yeah, I don't think it's _too_ confusing, but I do agree with Calvin that `.cacheElseNetwork`,`.cacheThenNetwork`, and `.networkElseCache` are clearer.

I prefer the clearer versions, but if we want to weigh parity of the naming across platforms higher than iOS team's preference on naming, I'm okay with that. I just want to make a decision today and move forward.

My feeling is that, while naming parity is nice, it's not something that we have consistently in other areas of the libraries currently. If we were aligned on naming in most of the library, then I'd try to stick with that, but not sure if there is much value in doing that here if we aren't consistent with it.

@bignimbus Do you have any opinion on this?